### PR TITLE
Document API save/load use cases

### PR DESCRIPTION
**Pull request by josvazg**
Add godoc source code documentation to help use the save and load functionality from the API directly.

### Collapsed
Screenshot from 2021-12-10 19-56-43

### Save
Screenshot from 2021-12-10 20-01-24

### Load
Screenshot from 2021-12-10 19-57-19

### CODE REVIEW

**Review by petewall — APPROVED** [review]

**Comment by migmartri** [comment]
Could you please add descriptions and links to the associated issues to your PRs please?

File: pkg/mover/doc_test.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 	"fmt"
 )
 
-// Package level documentation
+// Move a chart and its dependencies to another registry and or repository
```
Comment (migmartri) [inline pkg/mover/doc_test.go:10]: can't they be in the same file? Do we need 3 files for 3 examples?

File: pkg/mover/doc_load_test.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+func Example_load() {
+	// i.e ./mariadb-7.5.relocated.tgz
+	destinationPath := "%s-%s.relocated.tgz"
+
+	// Initialize the Mover action
+	chartMover, err := NewChartMover(
+		&ChartMoveRequest{
+			Source: Source{
+				Chart: ChartSpec{
+					// The source intermediate bundle path to place the charts and all its dependencies
```
Comment (migmartri) [inline pkg/mover/doc_load_test.go:21]: > // The source intermediate bundle path to place the charts and all its dependencies

the intermediate source tarball where the source chart and container images are present.

I'd also mention to see the Save example.

File: pkg/mover/doc_save_test.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	// So we get something like:
+	// $ tar tvf helm_chart.intermediate-bundle.tar
+	// -rw-r--r-- 0/0             201 1970-01-01 01:00 hints.yaml
+	// -rw-r--r-- 0/0             349 1970-01-01 01:00 original-chart/...
+	// -rw-r--r-- 0/0          773120 1970-01-01 01:00 images.tar
+	err = chartMover.Move()
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
```
Comment (migmartri) [inline pkg/mover/doc_save_test.go:47]: I'd add a to look into the load example

File: pkg/mover/doc_load_test.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+					IntermediateBundle: &IntermediateBundle{Path: "helm_chart.intermediate-bundle.tar"},
+				},
+				// no path to hints file as it is already coming inside the intermediate bundle
+				ImageHintsFile: "./image-hints.yaml",
```
Comment (migmartri) [inline pkg/mover/doc_load_test.go:25]: the hints file is not needed right?

**Review by migmartri — APPROVED** [review]
See inline comments

**Comment by migmartri** [comment]
Can you upload how the examples get showed in go doc?

**Comment by josvazg** [comment]
> Could you please add descriptions and links to the associated issues to your PRs please?

It was linked to https://github.com/vmware-tanzu/asset-relocation-tool-for-kubernetes/issues/15 already but did not add a close thing because this alone will not close the issue.

File: pkg/mover/doc_test.go (hunk 1 above)
Comment (josvazg) [reply pkg/mover/doc_test.go:10]: seems so:

https://go.dev/blog/examples
> Multiple examples can be provided for a given identifier by using a suffix beginning with an underscore followed by a lowercase letter. Each of these examples documents the Reverse function:
> 
> func ExampleReverse()
> func ExampleReverse_second()
> func ExampleReverse_third()

A single long code example would be confusing IMHO.

Comment (josvazg) [reply pkg/mover/doc_test.go:10]: Maybe I can put all in the same file and still have 3 separate samples.

Comment (josvazg) [reply pkg/mover/doc_test.go:10]: yep, that seems to work fine. Will be fixed in next update.

File: pkg/mover/doc_load_test.go (hunk 4 above)
Comment (josvazg) [reply pkg/mover/doc_load_test.go:25]: yep, I commented and then forgot to remove...

**Comment by josvazg** [comment]
> Can you upload how the examples get showed in go doc?

Added in the description.

File: pkg/mover/doc_test.go (hunk 1 above)
Comment (josvazg) [reply pkg/mover/doc_test.go:10]: done

**Pull request merged**